code/game/machinery/doors/brigdoors.dm

```diff
@@ -302,98 +297,74 @@
 /obj/machinery/door_timer/attack_hand(mob/user)
 	if(..())
 		return
-	interact(user)
-
-/obj/machinery/door_timer/interact(mob/user)
-	// Used for the 'time left' display
-	var/second = round(timeleft() % 60)
-	var/minute = round((timeleft() - second) / 60)
-
-	// Used for 'set timer'
-	var/setsecond = round((timetoset / 10) % 60)
-	var/setminute = round(((timetoset / 10) - setsecond) / 60)
-
-	user.set_machine(src)
-
-	// dat
-	var/dat = "<HR>Timer System:</hr>"
-	dat += " <b>Door [id] controls</b><br/>"
-
-	// Start/Stop timer
-	if(timing)
-		dat += "<a href='?src=[UID()];timing=0'>Stop Timer and open door</a><br/>"
-	else
-		dat += "<a href='?src=[UID()];timing=1'>Activate Timer and close door</a><br/>"
-
-	// Time Left display (uses releasetime)
-	dat += "Time Left: [(minute ? text("[minute]:") : null)][second] <br/>"
-	dat += "<br/>"
-
-	// Set Timer display (uses timetoset)
-	if(timing)
-		dat += "Set Timer: [(setminute ? text("[setminute]:") : null)][setsecond]  <a href='?src=[UID()];change=1'>Set</a><br/>"
-	else
-		dat += "Set Timer: [(setminute ? text("[setminute]:") : null)][setsecond]<br/>"
-
-	// Controls
-	dat += "<a href='?src=[UID()];settime=1'>Input Time</a>"
+	ui_interact(user)
+
+/obj/machinery/door_timer/ui_interact(mob/user, ui_key = "main", var/datum/nanoui/ui = null, var/force_open = 1)
+	ui = SSnanoui.try_update_ui(user, src, ui_key, ui, force_open)
+	if(!ui)
+		ui = new(user, src, ui_key, "brig_timer.tmpl", "Brig Timer", 500, 400)
```

Comment:
If this is really the window width please update your screenshots since it looks wayyy to wide in the screenshots

Reply:
Added one screenshot with new size, will add others as I make more corrections.